dodo.py

```diff
@@ -16,42 +16,92 @@
+
+
+    
+    """
+    Delete strings that aren't used in the code, to keep size down and not waste time fixing unused strings
+    Input: combined.properties.sorted
+    Output: filtered.properties
+    """
+    def filter():
+        REF_FILE = os.path.join("tabcmd", "locales", "codestrings.properties")
+        for current_locale in LOCALES:
+            LOCALE_PATH = os.path.join("tabcmd", "locales", current_locale)
+            IN_FILE = os.path.join(LOCALE_PATH, "LC_MESSAGES", "combined.properties")
+            OUT_FILE = os.path.join(LOCALE_PATH, "LC_MESSAGES", "filtered.properties")
+            
+            with open(REF_FILE, "r", encoding="utf-8") as ref:
+                required = ref.read()
+                with open(IN_FILE, "r", encoding="utf-8") as infile, open(OUT_FILE, "w+", encoding="utf-8") as outfile:
+                    for line in infile.readlines():
+                        key=line.split("=")[0]
+                        if key in required:
+                            outfile.writelines(line)
+                            
+                            
+            print("Filtered strings for {}".format(current_locale))
+    
+    """
+    For all languages: Read filtered.properties files with unicode like "Schlie\u00dfen", save it back as "Schließen"
     requires: pip install ftfy
     help: https://ftfy.readthedocs.io/
-    Run when we copy in updated properties files. NOT IDEMPOTENT: only run on the initial files
+    
+    Inputs: locales/*_[locale]/LC_MESSAGES/filtered.properties files
+    Output: (generated clean each run) locales/[locale]/LC_MESSAGES/transcoded.properties file
     """
 
-    def process_locales():
+    def encode():
         for current_locale in LOCALES:
-            if current_locale in ["en", "ga"]:  # greek is our pseudo-loc
-                continue
-            elif current_locale in ["ja", "ko", "zh"]:
-                encoding = "utf-8"
-            else:
-                encoding = "cp1252"
-
-            for file in glob.glob("tabcmd/locales/" + current_locale + "/*.properties"):
-                basename = os.path.basename(file).split(".")[0]
-                print("transcoding", basename)
-                with open(file, encoding=encoding) as infile:
+            # I'm not sure why we were varying the locale before, but this seems to work fine
+            encoding = "utf-8"
+
+            LOCALE_PATH = os.path.join("tabcmd", "locales", current_locale, "LC_MESSAGES")
+            INPUT_FILE = os.path.join(LOCALE_PATH, "filtered.properties")
+            OUTPUT_FILE = os.path.join(LOCALE_PATH, "transcoded.properties")
+            print("Encoding strings for " + current_locale)
+            try:
+                with open(INPUT_FILE, "r", encoding=encoding, errors='backslashreplace') as infile:
                     data = infile.read()
                 # now that we have read in the data properly encoded, fix the \u00fc characters and save as utf-8
```

Comment:
Does this comment still apply? I see `ü` used in the German file a bunch. Is it problematic?